homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -853,6 +894,26 @@ async def async_set_speed(self, speed: str) -> None:
             AirpurifierOperationMode[speed.title()],
         )
 
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the percentage of the fan."""
+        if self.supported_features & SUPPORT_SET_SPEED == 0:
+            return
+
+        # only values between 300 and 1200 are accepted
+        value = (percentage * 19) + 300
+
+        # value must be divisible by 10
+        if value % 10 != 0:
+            value = int((str(value))[:-1] + "0")
+
+        self._state_attrs[ATTR_MOTOR_SPEED] = percentage
```

Comment:
This doesn't match with the `percentage` property calculation.

Reply:
Is this better?

```suggestion
        self._state_attrs[ATTR_MOTOR_SPEED] = value
```

Reply:
Yes, that looks correct.

Reply:
@abrilevskiy could you commit this suggested change? I cannot write to your fork abrilevskiy:abrilevskiy-xiaomi-mb4